### PR TITLE
CORDA-1238: Moved the blob inspector out of experimental and wired it to JackonSupport

### PR DESCRIPTION
**Pull request by shamsasari**
The existing output format was not complete and so was deleted to avoid it becoming a tech debt. We can always resurrect it at a later point.

### CODE REVIEW

File: client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt [hunk 1]
```diff
@@ -274,38 +303,53 @@ object JacksonSupport {
 
     @Deprecated("This is an internal class, do not use")
     object PartySerializer : JsonSerializer<Party>() {
-        override fun serialize(value: Party, generator: JsonGenerator, provider: SerializerProvider) {
-            // TODO Add configurable option to output this as an object which includes the owningKey
-            generator.writeObject(value.name)
+        override fun serialize(value: Party, gen: JsonGenerator, provider: SerializerProvider) {
+            val mapper = gen.codec as PartyObjectMapper
+            if (mapper.isFullParties) {
```
Comment (fenryka) [inline client/jackson/src/main/kotlin/net/corda/client/jackson/JacksonSupport.kt]: feels like some syntax sugar would make this shorter given all branches are gen.writeObject(something)

File: client/jackson/src/test/kotlin/net/corda/client/jackson/JacksonSupportTest.kt [hunk 2]
```diff
@@ -386,7 +422,7 @@ class JacksonSupportTest(@Suppress("unused") private val name: String, factory:
     }
 
     @Test
-    fun X509Certificate() {
+    fun `X509Certificate serialisation`() {
```
Comment (fenryka) [inline client/jackson/src/test/kotlin/net/corda/client/jackson/JacksonSupportTest.kt]: serialization

File: serialization/src/main/kotlin/net/corda/serialization/internal/amqp/SerializationHelper.kt [hunk 3]
```diff
@@ -555,27 +558,28 @@ fun ClassWhitelist.hasAnnotationInHierarchy(type: Class<*>): Boolean {
...
-                        // and must be marked as both static and final (>0 means they're set)
-                        if (modifiers and Modifier.STATIC == 0 || modifiers and Modifier.FINAL == 0) null
-                        else {
-                            val accessibility = field.isAccessible
-                            field.isAccessible = true
-                            val obj = field.get(null)
-                            field.isAccessible = accessibility
-                            obj
-                        }
+fun Class<*>.objectInstance(): Any? {
```
Comment (fenryka) [inline serialization/src/main/kotlin/net/corda/serialization/internal/amqp/SerializationHelper.kt]: feels like a pointless refactor but *shrug*

File: tools/blobinspector/src/main/kotlin/net/corda/blobinspector/Main.kt [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        val bytes = source!!.readBytes().run {
+            require(size > amqpMagic.size) { "Insufficient bytes for AMQP blob" }
+            sequence()
+        }
+
+        require(bytes.take(amqpMagic.size) == amqpMagic) { "Not an AMQP blob" }
+
+        if (schema) {
+            val envelope = DeserializationInput.getEnvelope(bytes)
+            println(envelope.schema)
```
Comment (fenryka) [inline tools/blobinspector/src/main/kotlin/net/corda/blobinspector/Main.kt]: not printing transforms if present

File: docs/source/blob-inspector.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+Use the ``--help`` flag for a full list of command line options.
+
+``SerializedBytes`
+~~~~~~~~~~~~~~~~~~
+
+One thing to note is that the binary blob may contain embedded ``SerializedBytes`` objects. Rather than printing these
+out as a Base64 string, the blob inspector will first materialise them into Java objects and then output those. You will
+see this when dealing with classes such as ``SignedData`` or other structures that attach a signature, such as the node-info
+files or the ``network-parameters`` file in the node's directory. For example, the output of a node-info file may look like:
```
Comment (tlil) [inline docs/source/blob-inspector.rst]: Same as ^ (for consistency)

File: docs/source/blob-inspector.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+              "legalIdentitiesAndCerts" : [ "O=BankOfCorda, L=New York, C=US" ],
+              "platformVersion" : 4,
+              "serial" : 1527074180971
+            }
+          },
+          "signatures" : [ "dmoAnnzcv0MzRN+3ZSCDcCJIAbXnoYy5mFWB3Nijndzu/dzIoYdIawINXbNSY/5z2XloDK01vZRVTreFZCbZAg==" ]
+        }
+
+Notice the file is actually a serialised ``SignedNodeInfo`` object, which has a ``raw`` property of type ``SerializedBytes<NodeInfo>``.
+This property is materialised into the ``NodeInfo`` and is output under the ``deserialized`` field.
```
Comment (tlil) [inline docs/source/blob-inspector.rst]: `a ``NodeInfo`` object`

File: tools/blobinspector/src/main/kotlin/net/corda/blobinspector/Main.kt [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+import kotlin.system.exitProcess
+
+fun main(args: Array<String>) {
+    val main = Main()
+    try {
+        CommandLine.run(main, *args)
+    } catch (e: ExecutionException) {
+        val cause = e.cause!!
+        if (main.verbose) {
+            throw cause
```
Comment (tlil) [inline tools/blobinspector/src/main/kotlin/net/corda/blobinspector/Main.kt]: Controlled exit by using `cause.printStackTrace()` and `exitProcess(1)` instead?

File: docs/source/blob-inspector.rst [hunk 8]
```diff
@@ -0,0 +1,62 @@
+Blob Inspector
+==============
+
+There are many benefits to having a custom binary serialisation format (see :doc:`serialization` for details) but one
+disadvantage is the inability to view the contents in human-friendly manner. The blob inspector tool alleviates this issue
+by allowing the contents of a binary blob file (or URL end-point) to be output in either YAML or JSON. It uses
```
[the remark this answers is not in the document]
Comment (shamsasari) [reply docs/source/blob-inspector.rst]: Hmm, I think "format" is superfluous

File: docs/source/blob-inspector.rst [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+``java -jar blob-inspector.jar <file or URL>``
+
+Use the ``--help`` flag for a full list of command line options.
+
+``SerializedBytes`
+~~~~~~~~~~~~~~~~~~
+
+One thing to note is that the binary blob may contain embedded ``SerializedBytes`` objects. Rather than printing these
+out as a Base64 string, the blob inspector will first materialise them into Java objects and then output those. You will
+see this when dealing with classes such as ``SignedData`` or other structures that attach a signature, such as the node-info
```
[the remark this answers is not in the document]
Comment (shamsasari) [reply docs/source/blob-inspector.rst]: Not sure I agree. Elsewhere in the docs it's without the backticks - at least keep it consistent for now.

File: docs/source/blob-inspector.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (tlil) [reply docs/source/blob-inspector.rst]: Kk fair enough

File: tools/blobinspector/src/main/kotlin/net/corda/blobinspector/Main.kt (hunk 4 above)
Comment (shamsasari) [reply tools/blobinspector/src/main/kotlin/net/corda/blobinspector/Main.kt]: A separate PR I think. I'd need to see these transforms.

File: docs/source/blob-inspector.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (tlil) [reply docs/source/blob-inspector.rst]: It's only b/c it was done so for network params. See further down

**Comment by shamsasari** [comment]
Comments addressed

**Pull request merged**